test/libcommonserver/io/testcheckdirectoryiterator.cpp

```diff
@@ -0,0 +1,237 @@
+
+        DirectoryEntry entry;
+        CPPUNIT_ASSERT_EQUAL(false, it.next(entry, error));
+        CPPUNIT_ASSERT_EQUAL(IoError::IoErrorEndOfDirectory, error);
+    }
+
+    // Check that the directory iterator is valid when the path is a directory with one file and return the file on first call
+    {
+        const SyncPath directoryWithOneFile = _localTestDirPath / "test_dir_iterator/oneFile_dir";
+
+        IoError error;
+        DirectoryIterator it(directoryWithOneFile, false, error);
+        CPPUNIT_ASSERT_EQUAL(IoError::IoErrorSuccess, error);
+
+        DirectoryEntry entry;
+        CPPUNIT_ASSERT_EQUAL(true, it.next(entry, error));
+        CPPUNIT_ASSERT_EQUAL(IoError::IoErrorSuccess, error);
+
+
+        CPPUNIT_ASSERT_EQUAL(directoryWithOneFile / "testFile1.txt", entry.path());
+
+        CPPUNIT_ASSERT_EQUAL(false, it.next(entry, error));
+        CPPUNIT_ASSERT_EQUAL(IoError::IoErrorEndOfDirectory, error);
+    }
+
+    // Check that the directory iterator is valid when the path is a directory with one child directory
+    {
+        const SyncPath directoryWithOneChildDirectory = _localTestDirPath / "test_dir_iterator/oneDir_dir";
+        IoError error;
+        DirectoryIterator it(directoryWithOneChildDirectory, false, error);
+        CPPUNIT_ASSERT_EQUAL(IoError::IoErrorSuccess, error);
+
+        DirectoryEntry entry;
+        CPPUNIT_ASSERT_EQUAL(true, it.next(entry, error));
+        CPPUNIT_ASSERT_EQUAL(IoError::IoErrorSuccess, error);
+
+        CPPUNIT_ASSERT_EQUAL(directoryWithOneChildDirectory / "testDir1", entry.path());
+
+        CPPUNIT_ASSERT_EQUAL(false, it.next(entry, error));
+        CPPUNIT_ASSERT_EQUAL(IoError::IoErrorEndOfDirectory, error);
+    }
+
+
+    // Check that the directory iterator do not search recursively when the recursive flag is false
+    {
+        const SyncPath directoryWithMultipleSubDirectories = _localTestDirPath / "test_dir_iterator/recursive_dir";
+
+        IoError error;
+        DirectoryIterator it(directoryWithMultipleSubDirectories, false, error);
+        CPPUNIT_ASSERT_EQUAL(IoError::IoErrorSuccess, error);
+
+        DirectoryEntry entry;
+        for (int i = 0; i < 4; i++) {
+            CPPUNIT_ASSERT_EQUAL(true, it.next(entry, error));
+            CPPUNIT_ASSERT_EQUAL(IoError::IoErrorSuccess, error);
+        }
+
+        CPPUNIT_ASSERT_EQUAL(false, it.next(entry, error));
+        CPPUNIT_ASSERT_EQUAL(IoError::IoErrorEndOfDirectory, error);
+    }
+
+    // Check that the directory iterator search recursively when the recursive flag is true
+    {
+        const SyncPath directoryWithMultipleSubDirectories = _localTestDirPath / "test_dir_iterator/recursive_dir";
+
+        IoError error;
+        DirectoryIterator it(directoryWithMultipleSubDirectories, true, error);
+        CPPUNIT_ASSERT_EQUAL(IoError::IoErrorSuccess, error);
+
+        DirectoryEntry entry;
+        for (int i = 0; i < 36; i++) {
```

Comment:
Is it necessary to test it that many times?

Reply:
Could in fact be divided by 4. I changed it in next commit.